spark/src/main/scala/org/apache/spark/substrait/SparkTypeUtil.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.substrait
```

Comment:
Another minor thing I just noticed. Both this class and `ToSubstraitType` are under `org.apache.spark.substrait` instead of `io.substrait.spark` like everything else. Is that intentional?

Reply:
It appears that in `SparkTypeUtil`, the `sameType` method is accessing a private member of `org.apache.spark.sql.types.DataType` 🤮 so it has to be in the `org.apache.spark` namespace.
The other file though doesn't look like it needs to be there so I have moved it into `io.substrait.spark`.
If we're happy with this for now, I'll try and recode this bit at a later date.

Reply:
That's definitely 🤮, but we can improve that in the future. Thanks for looking into it.